lib/rules/no-wait-for-side-effects.ts

```diff
@@ -86,19 +135,26 @@ export default createTestingLibraryRule<Options, MessageIds>({
       }
     }
 
-    function reportImplicitReturnSideEffect(node: TSESTree.CallExpression) {
+    function reportImplicitReturnSideEffect(
+      node:
+        | TSESTree.CallExpression
+        | TSESTree.AssignmentExpression
+        | TSESTree.SequenceExpression
+    ) {
       if (!isCallerWaitFor(node)) {
         return;
       }
```

Comment:
```suggestion
      if (!isCallerWaitFor(node) || isRenderInImplicitReturnAssignment(node) || isRenderInSequenceAssignment(node)) {
        return;
      }
```

Reply:
I am afraid that won't work, as it can still be one of those:
```
        !helpers.isFireEventUtil(expressionIdentifier) &&
        !helpers.isUserEventUtil(expressionIdentifier) &&
        !helpers.isRenderUtil(expressionIdentifier)
```